src/main/kotlin/com/github/mitchellolsthoorn/testgenie/actions/GenerateTestsActionClass.kt

```diff
@@ -28,8 +33,14 @@ class GenerateTestsAction : AnAction() {
         val psiFile = e.dataContext.getData(CommonDataKeys.PSI_FILE)
         psiFile ?: return
 
+        val psiElement = e.dataContext.getData(CommonDataKeys.PSI_ELEMENT)
+
         val mainClass: PsiClass = PsiTreeUtil.findChildOfType(psiFile, PsiClass::class.java) ?: return
-        val classFQN = mainClass.qualifiedName ?: return
+        val classFileFQN = mainClass.qualifiedName ?: return
+
+        // Use FQN of the actually selected class (important in case of multiple classes in the same class file)
+        val classFQN = classFileFQN.substring(0, classFileFQN.lastIndexOf(".") + 1)
```

Comment:
You could maybe try casting `psiElement` to a class?
```  
if (psiElement instanceof PsiClass) 
```

Reply:
Good point! Indeed makes more sense. I have changed that